packages/kbn-apm-synthtrace/src/lib/logs/logs_synthtrace_es_client.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Client } from '@elastic/elasticsearch';
+import { dedot, ESDocumentWithOperation } from '@kbn/apm-synthtrace-client';
+import { pipeline, Readable, Transform } from 'stream';
+import { LogDocument } from '@kbn/apm-synthtrace-client/src/lib/logs';
+import { SynthtraceEsClient, SynthtraceEsClientOptions } from '../shared/base_client';
+import { getSerializeTransform } from '../shared/get_serialize_transform';
+import { Logger } from '../utils/create_logger';
+
+export type LogsSynthtraceEsClientOptions = Omit<SynthtraceEsClientOptions, 'pipeline'>;
+
+export class LogsSynthtraceEsClient extends SynthtraceEsClient<LogDocument> {
+  constructor(options: { client: Client; logger: Logger } & LogsSynthtraceEsClientOptions) {
+    super({
+      ...options,
+      pipeline: logsPipeline(),
+    });
+    this.dataStreams = ['logs-*-*'];
+  }
+}
+
+function logsPipeline() {
+  return (base: Readable) => {
+    return pipeline(
+      base,
+      getSerializeTransform<LogDocument>(),
+      getRoutingTransform(),
+      getDedotTransform(),
```

Comment:
consider whether you need to dedot - it's an expensive operation and you might not need it (yet)

Reply:
Good point, i will then comment out the call the moment until i figure out if we really need it for any specific fields